src/SB/Core/x/xEntBoulder.cpp

```diff
@@ -12,14 +38,167 @@ void xEntBoulder_FitToModel(xEntBoulder* ent)
     xVec3AddTo(&ent->bound.cyl.center, (xVec3*)&ent->model->Mat->pos);
 }
 
+void xEntBoulder_Render(xEnt* ent)
+{
+    xModelInstance* model = ent->model;
+    S32 shadowResult;
+    xVec3 shadVec;
+    F32 radius;
+
+    if ((model != NULL) && (xEntIsVisible(ent) != 0))
+    {
+        switch (model->Flags & 0x400)
+        {
+            case 0:
+                if (ent->flags & 0x40)
+                {
+                    if (iModelCull(model->Data, model->Mat) == 0)
+                    {
+                        xModelRender(model);
+                    }
+                }
+                else
+                {
+                    shadVec.x = model->Mat->pos.x;
+                    shadVec.y = model->Mat->pos.y - 10.0f;
+                    shadVec.z = model->Mat->pos.z;
+
+                    if (iModelCullPlusShadow(model->Data, model->Mat, &shadVec, &shadowResult) == 0)
+                    {
+                        xModelRender(model);
+                    }
+
+                    if (shadowResult == 0)
```

Comment:
Generally opt for `!truthyValue` instead of `truthyValue == 0/FALSE)` - it's more aligned with project conventions.